apps/web/app/api/(internal)/insights/lib/insights.ts

```diff
@@ -99,18 +98,41 @@ export const generateInsightsForSurveyResponsesConcept = async (surveyData: {
 
         skip += batchSize - responsesWithOpenTextAnswers.length;
 
-        responsesWithOpenTextAnswers.forEach((response) => {
-          openTextQuestionsWithInsights.forEach((question) => {
-            const responseText = response.data[question.id] as string;
-            if (responseText) {
-              const text = getPromptText(question.headline.default, responseText);
-              answersForDocumentCreation.push({
-                responseId: response.id,
-                questionId: question.id,
-                text,
-              });
-            }
-          });
+        const answersForDocumentCreationPromises = await Promise.all(
+          responsesWithOpenTextAnswers.map(async (response) => {
+            const attributes = response.personId ? await getAttributes(response.personId) : {};
+
+            return Promise.all(
```

Comment:
It appears that using `Promise.all` in this context might not be necessary because the map function is not returning promises, but rather objects or undefined and the `parseRecallInfo` and `getPromptText` functions seem to be synchronous operations. 🤔